### DISCUSSION TITLE
How to run nix-on-droid without proot in a rooted phone?

### DISCUSSION THREAD

**Discussion by nilp0inter** [nix-community/nix-on-droid#249] [Q&A]
Hi,

I've been using `nix-on-droid` for a couple of years, and I am delighted by how well it works and how neat it is to carry my Nix setup everywhere I go.

I am about to flash a custom ROM in my phone. And I was wondering if it would be possible to run `nix-on-droid` (or maybe just plain Nix?) or a subset of it in a rooted phone without `proot` for a boost in performance.

Any thoughts, advice or opinions?

Cheers,
Roberto

**Comment by Djspac3** [comment]
Uhh I don't think this is coming until this is stable
I'm not a dev but first it would need to tell if you have a magisk, kernalsu, or some other root method and check for like busybox and then ask for permission and then check THEN finally run a lot more customized commands
